cost estimation

choldgraf opened the pull request:
This is a first pass at adding a cost estimation demo. Eventually we'd probably want this to be a proper "widget", but for now it's an IFrame serving a binder instance (nice self-promotion here :-D ).

See it live here: http://predictablynoisy.com/zero-to-jupyterhub-k8s/cost.html#interactive-estimator

We should discuss how to improve / parameterize the cost estimation widget. Maybe @yuvipanda and I can get a good first iteration done today?

ryanlovett commented:
This is awesome!

Just one suggestion is to parameterize the timeframe for instructors who want to extrapolate based on a single day or week's curve.

choldgraf commented:
@ryanlovett check out the website now! :)

choldgraf commented:
@yuvipanda I think it's almost ready to go. If you wanna comment w/ some better numbers when you get a chance, I can update and then we can merge!

choldgraf commented:
also, a simple autoscaling function is added in now :)

yuvipanda commented:
I don't want us to iframe binder yet tho - we don't have efficient culling
yet on binder, and there's a bunch of these pods running there for now :D
I'd like to hold off on the iframe for a week or two at least - we can make
it click to launch till then?

On Thu, Jun 1, 2017 at 4:41 PM, Chris Holdgraf <notifications@github.com>
wrote:

> also, a simple autoscaling function is added in now :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/82#issuecomment-305649961>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB23lW2KpdjuOe1bxlDW-WbzLl_JCdbks5r_0wcgaJpZM4NtY-C>
> .
>



-- 
Yuvi Panda T
http://yuvi.in/blog

choldgraf commented:
works for me, I can remove the iframe when I update the cost estimates

choldgraf commented:
I think that we should make a modification to this before it gets merged. In particular, we can take whatever parameters people put in right now, and use them to choose a "best-fit machine" for those parameters (e.g., RAM / CPU reqs). Then we can calculate the cost from this page as an example:

https://cloud.google.com/compute/docs/machine-types

choldgraf commented:
Heya folks - I spent some time today updating the cost estimation page since I feel like it's something people ask for often and I'd like to get a decent-enough version working before JupyterCon.

I've updated the jupyter notebook so that it has you choose from several machines instead of specifying CPU/RAM directly, also changed the `iframe` approach so that we use a button instead.

http://predictablynoisy.com/zero-to-jupyterhub-k8s/cost.html

LMK what you think, in particular let me know if those numbers look right!

choldgraf commented:
maybe @yuvipanda and @willingc have thoughts on this!

willingc commented:
@choldgraf Nice :-)

A few usability thoughts:
- Mention that pressing the button will open the Binder service which will serve you the interactive notebook.
- add a disclaimer re: guidance. Check with your cloud provider.

choldgraf commented:
good ideas! I added some extra language on this in the latest push.

choldgraf commented:
wohoo! self merge!
